### PR TITLE
Rebuild with hdf5 1.12.1 on all platforms

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Rebuild with hdf5 1.12.1 on all platforms to have consistent dependencies. Also cleaned up dependencies.

### CODE REVIEW

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,9 +9,10 @@ source:
   sha256: 34f3fa2366ce20b18f1df573a77c1d27306ce1f2a41d9f9eff621b5192ea8788
   patches:
     - patches/use_system_blosc2.patch
+    - patches/0001-Don-t-copy-blosc2-in-the-package.patch
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:12]: why we do not want to copy it ?

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGMT, but can you explain the reason why we do not want to copy blosc2 ?

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:12]: Because we should not vendor dependencies that we already have on defaults.

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:12]: yes, I can see now we have blosc-feedstock

**Pull request merged**